### ISSUE TITLE
Apply test config profile (%test properties) for runtime properties when running integration tests (native tests)

### ISSUE THREAD

**Issue by yrodiere** [quarkusio/quarkus#26279]
### Description

When running integration tests (`@QuarkusIntegrationTest`), we execute tests against a native build of the application, one that was built with the `prod` configuration profile. The main reason for that is that a native build takes a lot of time, and doing two native builds of the application (once with the `prod` profile, once with the `test` profile`) would add unreasonable overhead.

This sometimes leads to confusing behavior, e.g. some configuration properties such as `%test.quarkus.hibernate-orm.sql-load-script` or  `%test.quarkus.hibernate-search-orm.automatic-indexing.synchronization.strategy` being ignored in native integration tests.

While I think this limitation is acceptable for build-time properties, I think we would get much better user experience by at least applying *runtime* `%test` properties when running (native) integration tests, as most people probably expect.

Technically it should be doable as it wouldn't require to re-build the application, and it would certainly make configuration more intuitive.

While we're at it, we could even consider displaying warnings when starting the (native) integration tests if the developer tried to set build-time properties in the `%test` profile.

**Comment by yrodiere** [comment]
> In my view, we should simply add `quarkus.test.intergration-test-profile` and make things very clear in the documentation

And probably have it default to `test`, then? Otherwise I fear we'll keep confusing people... Documentation is nice, but "instinctive" behavior is better.

**Comment by geoand** [comment]
> And probably have it default to test, then?

That's where I disagree :)

**Comment by geoand** [comment]
I don't agree that it's bad default, although I understand why people (perhaps even most?) would think so.

The way I see it, the integration test should do exactly the same thing as the application run in production, unless otherwise requested.
As for the argument that people would need to use  `prod` configuration, well I'm totally fine with that because people should never have **real** production configuration in `application.properties` anyway.

**Comment by geoand** [comment]
Yeah, I understand the reasoning for those specific properties, although I still don't think it's worth changing the existing behavior.

In any case, if someone makes the change, I personally won't block it, but I won't be doing the work myself (although I can provide pointers to what needs to be changed).

**Comment by geoand** [comment]
I am going to close this in favor of https://github.com/quarkusio/quarkus/issues/24581 which we had discussed at some point.

**Comment by yrodiere** [comment]
I doubt this solution will do much to reduce confusion from users who set a configuration property in the "test" profile and don't see it applied when running tests.

But as I stated above, I agree it's a start:

> I'm just guessing since you didn't give any details, but if your point is that integration tests should use different configuration than other tests, then at the very least we'd need quarkus.test.integration-test-profile to default to a dedicated profile, e.g. integrationtest. Together with https://github.com/quarkusio/quarkus/issues/26283, I suppose people would then be able to configure both tests and integration tests easily (e.g. %[test,integrationtest,dev].quarkus.datasource.jdbc.url=jdbc:postgresql://localhost/quarkus_test).
